### PR TITLE
test: extend view filtering test case

### PR DESCRIPTION
**Pull request by psarna**
In order to cover more code paths, the test case
now places filtering on various combinations of base columns,
including both primary keys and regular columns.
It also makes the test scylla_only, as filtering is an extension
not supported in Cassandra right now.

### CODE REVIEW

File: test/cql-pytest/test_materialized_view.py [hunk 1]
```diff
@@ -196,13 +196,20 @@ def test_mv_is_not_null(cql, test_keyspace):
...
     with new_test_table(cql, test_keyspace, schema) as table:
-        with new_materialized_view(cql, table, select='p,c,v', pk='p,c,v', where='p IS NOT NULL and c IS NOT NULL and v = 44') as mv:
-            cql.execute(f"INSERT INTO {table} (p,c,v) VALUES (42,43,44)")
-            cql.execute(f"INSERT INTO {table} (p,c,v) VALUES (1,2,3)")
-            assert list(cql.execute(f"SELECT * FROM {mv}")) == [(42, 43, 44)]
+        for p_condition in ['p = 42', 'p IS NOT NULL']:
+            for c_condition in ['c = 43', 'c IS NOT NULL']:
+                for v_condition in ['v = 44', 'v IS NOT NULL']:
+                    where = f"{p_condition} AND {c_condition} AND {v_condition}"
+                    with new_materialized_view(cql, table, select='p,c,v', pk='p,c,v', where=where) as mv:
```
Comment (nyh) [inline test/cql-pytest/test_materialized_view.py:211]: There's an interesting feature of this test that didn't exist in the previous version, that view building (of the pre-existing data) races with the write of the new (and identical) data. It shouldn't change anything in the results, but has the potential of catching unrelated bugs (which I guess, isn't bad).

File: test/cql-pytest/test_materialized_view.py [hunk 2]
```diff
@@ -196,13 +196,20 @@ def test_mv_is_not_null(cql, test_keyspace):
...
 # which erroneously also includes static columns if such are present in the
 # base table. Currently views only operate on regular columns and the filtering
-# code assumes that. TODO: once we implement static column support for materialized
+# code assumes that. Once we implement static column support for materialized
 # views, this test case will be a nice regression test to ensure that everything still
 # works if the static columns are *not* used in the view.
-def test_filter_with_unused_static_column(cql, test_keyspace):
+# This test goes over all combinations of filters for partition, clustering and regular
+# base columns.
+def test_filter_with_unused_static_column(cql, test_keyspace, scylla_only):
```
Comment (nyh) [inline test/cql-pytest/test_materialized_view.py:204]: If I understand correctly, the "scylla_only" is only for the v=44 case?
I guess it's ok. This test doesn't need to be the place where we document what is the exact difference between Scylla and Cassandra. But if we did want to do it, we could create two separate tests - one for the v = 44 case and the second for the others (which also works on Cassandra).

**Review by nyh — APPROVED** [review]

File: test/cql-pytest/test_materialized_view.py (hunk 2 above)
Comment (psarna) [reply test/cql-pytest/test_materialized_view.py:204]: Yes, I just decided not to split into two cases just for the sake of testing Cassandra on this path, since the semantics of view filtering are quite clear.

File: test/cql-pytest/test_materialized_view.py (hunk 1 above)
Comment (psarna) [reply test/cql-pytest/test_materialized_view.py:211]: Yes, initially I added a "truncate {table}" request to the end of the loop body, but then I realized that it's enough that a new synchronous update arrives, so I minimized the test case a little

**Comment by psarna** [comment]
Oh, I tested that too, I just forgot to post results. Here's what returned before the fix, for each tested where clause:
```
test/cql-pytest/test_materialized_view.py 
[p = 42 AND c = 43 AND v = 44]: FAILED
[p = 42 AND c = 43 AND v IS NOT NULL]: [Row(p=42, c=43, v=44)], expected: [(42, 43, 44)]
[p = 42 AND c IS NOT NULL AND v = 44]: FAILED
[p = 42 AND c IS NOT NULL AND v IS NOT NULL]: [Row(p=42, c=43, v=44)], expected: [(42, 43, 44)]
[p IS NOT NULL AND c = 43 AND v = 44]: FAILED
[p IS NOT NULL AND c = 43 AND v IS NOT NULL]: [Row(p=42, c=43, v=44)], expected: [(42, 43, 44)]
[p IS NOT NULL AND c IS NOT NULL AND v = 44]: FAILED
[p IS NOT NULL AND c IS NOT NULL AND v IS NOT NULL]: [Row(p=42, c=43, v=44), Row(p=1, c=2, v=3)], expected: [(42, 43, 44), (1, 2, 3)]
```
tl;dr: regular base columns are problematic

**Comment by psarna** [comment]
Make sense, too: with wildcard selection, the order is: primary key columns, static columns, regular columns. So the order of key restrictions would remain the same when stray static columns were present, but regular ones would have issues
